fuels-signers/src/wallet.rs

```diff
@@ -0,0 +1,264 @@
+        match witnesses.len() {
+            0 => tx.set_witnesses(new_witnesses),
+            _ => {
+                witnesses.extend(new_witnesses);
+                tx.set_witnesses(witnesses)
+            }
+        }
+    }
+
+    /// Transfer funds from this wallet to another `Address`.
+    /// Fails if amount for color is larger than address's spendable coins.
+    ///
+    /// # Examples
+    /// ```
+    /// use fuels_signers::provider::Provider;
+    /// use fuels_signers::{LocalWallet, Signer};
+    /// use fuels_signers::util::test_helpers::{
+    ///     setup_address_and_coins, setup_local_node, setup_test_provider,
+    /// };
+    /// use fuel_tx::{Bytes32, Color, Input, Output, UtxoId};
+    /// use rand::{rngs::StdRng, RngCore, SeedableRng};
+    /// use secp256k1::SecretKey;
+    /// use std::str::FromStr;
+    ///
+    /// async fn foo() -> Result<(), Box<dyn std::error::Error>> {
+    ///   // Setup test wallets with 1 coin each
+    ///   let (pk_1, mut coins_1) = setup_address_and_coins(1, 1);
+    ///   let (pk_2, coins_2) = setup_address_and_coins(1, 1);
+    ///   coins_1.extend(coins_2);
+    ///
+    ///   // Setup a provider and node with both set of coins
+    ///   let provider = setup_test_provider(coins_1).await;
+    ///
+    ///   // Create the actual wallets/signers
+    ///   let wallet_1 = LocalWallet::new_from_private_key(pk_1, provider.clone()).unwrap();
+    ///   let wallet_2 = LocalWallet::new_from_private_key(pk_2, provider).unwrap();
+    ///
+    ///   // Transfer 1 from wallet 1 to wallet 2
+    ///   let _receipts = wallet_1
+    ///        .transfer(&wallet_2.address(), 1, Default::default())
+    ///        .await
+    ///        .unwrap();
+    ///
+    ///   let wallet_2_final_coins = wallet_2.get_coins().await.unwrap();
+    ///
+    ///   // Check that wallet two now has two coins
+    ///   assert_eq!(wallet_2_final_coins.len(), 2);
+    ///   Ok(())
+    /// }
+    /// ```
+    pub async fn transfer(
+        &self,
+        to: &Address,
+        amount: u64,
+        color: Color,
+    ) -> io::Result<Vec<Receipt>> {
+        let spendable = self.get_spendable_coins(&color, amount).await?;
+
+        let mut inputs: Vec<Input> = vec![];
+        let outputs: Vec<Output> = vec![
+            Output::coin(*to, amount, color),
+            Output::change(self.address(), 0, color),
+        ];
+
+        for coin in spendable {
+            let input_coin = Input::coin(
+                UtxoId::from(coin.utxo_id),
+                coin.owner.into(),
+                coin.amount.0,
+                color,
+                0,
+                0,
+                vec![],
+                vec![],
+            );
+
+            inputs.push(input_coin);
+        }
+
+        // Build transaction and sign it
+        let mut tx = self.provider.build_transfer_tx(&inputs, &outputs);
+        let sig = self.sign_transaction(&tx).await.unwrap();
+        self.append_signatures(&mut tx, &[sig]);
```

Comment:
Would it make sense to combine these into one API?

Reply:
You mean once you `sign_transaction(&tx)` it would mutate `&tx` by appending the sign to it?

Reply:
Yep exactly. It would simplify the most common usecases. We could expose the underlying keystore signature api for arbitrary data if people need that later.

Reply:
That's a good idea!

Reply:
I don't think this is a blocker though.

Reply:
I did it now anyway, it was a pretty simple change.